bench/process/process.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/sh
+# shellcheck disable=SC1090
+set -eu
+
+fail() {
+    echo -e "ERROR:  $*" >&2
+    exit 1
+}
+
+default_op='report'
+default_format='csv'
+
+format=$default_format
+header_footer='false'
+no_progress=
+
+while test $# -ge 1
+do case "$1" in
+       --github )          format='github';;
+       --csv )             format='csv'; header_footer='false';;
+       --cls )             echo -en "\ec" >&2;;
+       --no-progress )     no_progress='true';;
+       --trace )           set -x;;
+       * )                 break;; esac; shift; done
+
+op=${1:-$default_op}; shift || true
+
+main() {
+    PROCESS_ORIG_PWD=$(pwd)
+    pushd "$(dirname $0)" >/dev/null || exit 1
+    case "${op}" in
+        collect )    op_collect "$@";;
+        process )    op_process;;
+        render )     op_render;;
+
+        report )     op_collect "$@" | op_process | op_render;;
+
+        call )       eval "$@";;
+        * ) echo "ERROR:  operation must be one of:  collect process render report" >&2; exit 1;; esac
+}
+
+hardcoded_branch='membench'
+hardcoded_commit='a7ee17d1af44b571c6e476916bd24ed65db97e15'
```

Comment:
Why is it necessary to put it here? Seams to me that it breaks the abstract nature of the other part of the script.

Reply:
Oh, yes, I've gone out of my way to make it painfully obvious so I have a chance to do it later.

Perhaps later has finally occured! : -)